src/shared/watchedFiles.ts

```diff
@@ -93,6 +97,26 @@ export abstract class WatchedFiles<T> implements vscode.Disposable {
         await this.rebuild()
     }
 
+    /**
+     * Create a special watcher that operates only on untitled files.
+     * VSCode does not make it easy to watch untitled files and globs won't work without knowing exactly
```

Comment:
"Watching" an in-memory file is just listening for the vscode `didChange` event (as done here). I wouldn't expect filesystem watchers to be involved. So the comment is kind of confusing.

```suggestion
     * To "watch" the in-memory contents of an untitled:/ file we just subscribe to `onDidChangeTextDocument`.
```